### PR TITLE
openstack: parallelize creation of PVCs

### PR DESCRIPTION
**Pull request by bennyz**
Currently the BeforeTransferHook waits for all images to be ready before creating the PVCs. However, some images will be slower than other to create, so instead we create the PVC as soon as the image is ready to save some time.

### CODE REVIEW

File: pkg/controller/plan/migration.go [hunk 1]
```diff
@@ -593,31 +593,37 @@ func (r *Migration) execute(vm *plan.VMStatus) (err error) {
 			vm.AddError(fmt.Sprintf("Step '%s' not found", r.step(vm)))
 			break
 		}
+		var pvcNames []string
```
Comment (ahadas) [inline pkg/controller/plan/migration.go:596]: why moving it so far from where it's used?

File: pkg/controller/plan/migration.go [hunk 2]
```diff
@@ -593,31 +593,37 @@ func (r *Migration) execute(vm *plan.VMStatus) (err error) {
...
+			r.Log.Info("BeforeTransfer hook isn't ready yet")
+			return
+		}
+
+		err = r.kubevirt.createPodToBindPVCs(vm, pvcNames)
+		if err != nil {
+			step.AddError(err.Error())
+			err = nil
+			break
+		}
```
Comment (ahadas) [inline pkg/controller/plan/migration.go:626]: maybe it's a result of rebase conflict - it seems incorrect, did you intentionally pushed it out from the `if r.kubevirt.isOpenstack(vm)` scope?

Comment (bennyz) [reply pkg/controller/plan/migration.go:626]: yeah, it's not intentional, I forgot to wrap it again

File: pkg/controller/plan/migration.go [hunk 3]
```diff
@@ -599,17 +599,17 @@ func (r *Migration) execute(vm *plan.VMStatus) (err error) {
 			err = liberr.Wrap(err)
 			return
 		}
-		if !ready {
```
[the remark this answers is not in the document]
Comment (ahadas) [reply pkg/controller/plan/migration.go:602]: maybe beforeTransferActions?

[the remark this answers is not in the document]
Comment (bennyz) [reply pkg/controller/plan/migration.go:602]: renamed to PreTransferActions before I saw this comment, close enough 😅

[the remark this answers is not in the document]
Comment (ahadas) [reply pkg/controller/plan/migration.go:602]: and I actually thought about replacing "before" with "pre" :)

**Review by ahadas — APPROVED** [review]

**Pull request merged**